release_notes/ocp_3_6_release_notes.adoc

```diff
@@ -0,0 +1,2576 @@
+
+`SecurityContextConstraints` are now also available via a groupified API at
+*_/apis/security.openshift.io/v1/securitycontextconstraints_*. They are still
+available at *_/api/v1/securitycontextconstraints_*, but using the groupified API
+will provide better integration with tooling.
+
+[discrete]
+[[ocp-36-volume-recycler-now-deprecated]]
+=== Openshift Volume Recycler Now Deprecated
+
+Openshift Volume Recycler is being deprecated. Anyone using recycler should use
+dynamic provision and volume deletion instead.
+
+[[ocp-36-bug-fixes]]
+== Bug Fixes
+
+This release fixes bugs for the following components:
+
+*Authentication*
+
+* Nested groups now sync between {product-title} and Active Directory. It is
+common to have nested groups in Active Directory.  Users wanted to be able to
+sync such groups with {product-title}. This feature was always supported, but
+lacked any formal documentation and was difficult to discover.
+xref:../install_config/syncing_groups_with_ldap.adoc#sync-ldap-nested-example[Documentation
+is now added].
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1437324[*BZ#1437324*])
+
+*Builds*
+
+* When a build is started from a webhook, the server response does not contain a
+body Therefore, the CLI cannot easily determine the generation of the created
+build, and cannot report it to the user. Change webhook response to contain the
+created build object in the body. The CLI can now report the correct build
+generation when created.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1373441[*BZ#1373441*])
+
+* Build durations are recorded as part of a storage hook. Build duration is
+sometimes calculated incorrectly and reported with an invalid value. Calculate
+build duration when recording build time of build completion. As a result, build
+durations are reported correctly and align with the build start and completion
+times. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1443687[*BZ#1443687*])
+
+* The code was not setting the status reason and status message for certain
+failures. Therefore, there were missing status reasons and status messages for
+certain failures. With this bug fix, code was added that sets the status reason
+and status message and the status reason and message are now set.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1436391[*BZ#1436391*])
+
+* A debug object type is used when high levels of logging are requested. Client
+code did not anticipate the alternative object type and experienced a typecast
+error. With this bug fix, the client code is updated to handle the debug object
+type. The typecast error will not occur and builds now proceed as expected.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1441929[*BZ#1441929*])
+
+* When resources were specified in the build default configuration, the resource
+values were not applied to the build pods. They were only applied to the build
+object. Builds ran without the default resource li
```

Comment:
we just had a new one discovered today unfortunately. 

When performing builds using an image source input, directories within the input content are injected with permissions of 0700 with the default image user as the owner.  This means the content is unlikely to be accessible when the application image is run under a random uid.  This can be worked around by performing a chmod operation in either the assemble script (for s2i builds) or the Dockerfile (for Docker builds).  Most OpenShift S2I builder images already perform this chmod operation, but custom built s2i builder images or builds using custom assemble scripts may not.